internal/indexer/indexer.go

```diff
@@ -723,6 +813,115 @@ func (i *Indexer) emitContainsForProject() {
 	}
 }
 
+func (i *Indexer) indexPackageDeclarations() {
+	i.visitEachPackage("Indexing packages", i.indexPackageDeclarationForPackage)
+}
+
+func newPkgDeclaration(p *packages.Package, f *ast.File) (*PkgDeclaration, token.Position) {
+	pkgKeywordPosition := p.Fset.Position(f.Package)
+
+	name := f.Name.Name
+
+	// TODO(packages): Is this the right package information here? I'm not sure.
+	// obj := types.NewPkgName(f.Package, p.Types, name, types.NewPackage(p.PkgPath, name))
+	return &PkgDeclaration{
```

Comment:
Do you have to construct a type here? Can't you get it from `f.Syntax` somewhere?

Reply:
It has no f.Syntax type, it is a conglomeration of fields from `f.` that tells us the information. For example, the position is just stored in `f.Package`.